### ISSUE TITLE
Expo dev client crash on fast refresh

### ISSUE THREAD

**Issue by mozzius** [mrousavy/react-native-mmkv#566]
`"react-native-mmkv": "2.10.1"`
`"expo": "49.0.7"`

I'm using Expo dev client.

Works completely fine on first load, then when you refresh (for instance by pressing `r` in the terminal) I get:

```
Error: Failed to create a new MMKV instance, the native initializer function does not exist. Are you trying to use MMKV from different JS Runtimes?, js engine: hermes
```

Happens consistantly every time

**Comment by SohelIslamImran** [comment]
I'm facing this error on android

```sh
Error: Exception in HostFunction: Failed to create MMKV instance!, js engine: hermes
```

**Comment by mrousavy** [comment]
@SohelIslamImran stop spamming.

**Comment by chrisdrackett** [comment]
for me this isn't only on fast refresh, but on every subsequent load of the app after the initial (working) load.

**Comment by FlemingVincent** [comment]
@chrisdrackett Same for me

**Comment by mrousavy** [comment]
1. Can you reproduce the issue in a fresh app that only has react-native-mmkv installed?
2. I believe this is related to react-native-reanimated.

**Comment by chrisdrackett** [comment]
in case it helps anyone else this is the test app I'm using. So far I've added `react-native-reanimated` and I am not seeing the crash. I'll try and move it closer to my current project looking for what causes it when I have more time.

https://github.com/chrisdrackett/mmkv-49

**Comment by chenzesam** [comment]
Same issue, any new idea?

**Comment by oscar-fk** [comment]
In my case, this is solved by deleting node_modules, running yarn again, and then doing a new expo build.

**Comment by chrispader** [comment]
Unfortunately i cannot reproduce this. Can anyone who's still experiencing this give me his/her lockfile and app's dependencies?

**Comment by nmalzieu** [comment]
> Unfortunately i cannot reproduce this. Can anyone who's still experiencing this give me his/her lockfile and app's dependencies?

Hey Chris. Here's my yarn.lock
I still have the issue, every day! Only in expo dev client, when using fast refresh.

Happy to give you more insight anytime

[yarn.lock.txt](https://github.com/mrousavy/react-native-mmkv/files/13310119/yarn.lock.txt)

**Comment by chrispader** [comment]
Thanks!

**Comment by chrispader** [comment]
thanks for your help guys!

**Comment by mrousavy** [comment]
Fixed in 2.11.0! ❤️💪
